src/Security/src/Microsoft.AspNetCore.Authentication.OpenIdConnect/ClientAuthentication/FormPost.cs

```diff
@@ -0,0 +1,16 @@
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Authentication.OpenIdConnect
+{
+    /// <summary>
+    /// Send client id and client secret in the request body 
+    /// </summary>
+    public class FormPost : IClientAuthentication
+    {
+        public HttpRequestMessage SetClientAuthentication(HttpRequestMessage message, OpenIdConnectMessage tokenEndpointRequest)
+        {
+            return message;
```

Comment:
This factoring is a bit odd when only one of the two actually implements it's logic here. This also doesn't look like a point that's likely to be extended in the future. Why not simplify this down to an enum and an if statement in OIDCHandler? It would only be about 7 lines of code.

Reply:
There is other's authentication mode that's why. I agree FormPost is a bit odd but didn't went to change every thing to have the parameter in it.